src/NuGet.Core/NuGet.Versioning/VersionRangeFormatter.cs

```diff
@@ -185,7 +185,11 @@ private string GetNormalizedString(VersionRange range)
 
             sb.Append(range.HasUpperBound && range.IsMaxInclusive ? ']' : ')');
 
-            return sb.ToString();
+            string result = sb.ToString();
+
+            StringBuilderPool.Shared.Return(sb);
+
+            return result;
```

Comment:
This pattern is very common. What would folks say about changing `Return` to also do the `ToString` so we end up with this instead?

```suggestion
            return StringBuilderPool.Shared.Return(sb);
```

The method could be renamed to `ToStringAndReturn` or something if that'd be clearer. Looking through the code, all instances of `Return` could be changed to this. There's one in a `finally` block, which I think could be inlined (which would help perf a little too.)

Reply:
Yep sounds good, I probably added this in the first place?

Reply:
Yes, you added it in 680f9bd4e97db7cd7482584276886764de69d3cb.

I pushed this change in an update.